lib/paperclip/thumbnail.rb

```diff
@@ -119,13 +133,25 @@ def animated?
     # Return true if ImageMagick's +identify+ returns an animated format
     def identified_as_animated?
       if @identified_as_animated.nil?
-        @identified_as_animated = ANIMATED_FORMATS.include? identify("-format %m :file", file: "#{@file.path}[0]").to_s.downcase.strip
+        @identified_as_animated = ANIMATED_FORMATS.include? identified_as
       end
       @identified_as_animated
     rescue Terrapin::ExitStatusError => e
       raise Paperclip::Error, "There was an error running `identify` for #{@basename}" if @whiny
     rescue Terrapin::CommandNotFoundError => e
       raise Paperclip::Errors::CommandNotFoundError.new("Could not run the `identify` command. Please install ImageMagick.")
     end
+
+    def identified_as_apng?
+      if @identified_as_apng.nil?
+        @identified_as_apng = File.extname(@file.path) == '.png' || identified_as == 'png'
+        @identified_as_apng &&= identify("-format %m apng::file", file: @file.path).to_s.downcase.strip.match?(/\A(apng){2,}/)
+      end
+      @identified_as_apng
+    end
+
+    def identified_as
+      @identified_as ||= identify("-format %m :file", file: "#{@file.path}[0]").to_s.downcase.strip
```

Comment:
i believe this is just simplifying the old code, line 122

Reply:
The method is also called from `dentified_as_apng?`.

Reply:
Or perhaps I misunderstand your concern. Do you suggest that I change this?